plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java

```diff
@@ -49,23 +67,84 @@ public List<CredentialType> getAvailableCredentialTypes() {
     @Override
     public List<AbstractCredentialInfo> getCredentialDefinitions() {
         List<AbstractCredentialInfo> credentialInfos = new ArrayList<>();
-
         List<CredentialVariable> credentialVariables = new ArrayList<>();
-        CredentialDefinition accessKey = new CredentialDefinition("AKSK",
+        CredentialDefinition accessKey = new CredentialDefinition(HuaweiCloudMonitorConstants.IAM,
                 "The access key and security key.",
                 CredentialType.VARIABLES, credentialVariables);
-        credentialVariables.add(new CredentialVariable("HW_ACCESS_KEY", "The access key."));
-        credentialVariables.add(new CredentialVariable("HW_SECRET_KEY", "The security key."));
-
+        credentialVariables.add(
+                new CredentialVariable(HuaweiCloudMonitorConstants.HW_ACCESS_KEY,
+                        "The access key."));
+        credentialVariables.add(
+                new CredentialVariable(HuaweiCloudMonitorConstants.HW_SECRET_KEY,
+                        "The security key."));
         credentialInfos.add(accessKey);
 
         return credentialInfos;
     }
 
     @Override
     public List<Metric> getMetrics(AbstractCredentialInfo credential,
-            DeployResource deployResource) {
-        return null;
+            DeployResource deployResource, MonitorResourceType monitorResourceType) {
+        List<Metric> metrics = new ArrayList<>();
+        try {
+            clearExpiredMetricCache(deployResource.getResourceId());
```

Comment:
since we return data for the past "x" minutes only.. we must have this documented in the interface. If this, could be different for each cloud, we must document it accordingly in the API.

Reply:
We do not fetch data from the past period of time, we obtain real-time data. However, when a scheduled task calls HUAWEI CLOUD monitoring data, if the call is frequent, the interface will limit the flow, so there will be a buffer solution. When the HUAWEI CLOUD monitoring interface reports an error , we will fetch the data in the cache, and the monitoring data of each vm in the cache will be invalidated in 30 minutes, but these invalidated data will be cleared every 12 hours